blocks/gu_spoverview/block_gu_spoverview.php

```diff
@@ -45,92 +48,134 @@ public function has_config() {
-            'assess_marked_str'             => $assess_str.get_string('marked', $tpl),
+            'assignment_submitted_count' => sizeof($assignsubmitted),
+            'to_be_submitted'            => (int) sizeof($assignments) - (int) sizeof($assignsubmitted),
+            'due_date'                   => sizeof($assignmentduedate),
+            'marked_assessment_count'    => sizeof($assessments),
+            'assignment_svg_path'        => '../blocks/gu_spoverview/pix/assignment-submitted.svg',
+            'to_be_submitted_svg_path'   => '../blocks/gu_spoverview/pix/to-be-submitted.svg',
+            'overdue_svg_path'           => '../blocks/gu_spoverview/pix/overdue.svg',
+            'assessment_marked_svg_path' => '../blocks/gu_spoverview/pix/assessment-marked.svg'
         ];
 
         $this->content = new stdClass;
+
         $this->content->text = $OUTPUT->render_from_template('block_gu_spoverview/spoverview', $templatecontext);
 
         return $this->content;
     }
 
+    public static function retrieve_assessment($courseids, $activities, $multipleassessment, $userid){
+        global $DB;
+
+        $assessments = array();
+        list($inactivities, $aparams) = $DB->get_in_or_equal($activities, SQL_PARAMS_NAMED);
+        list($incourses, $cparams) = $DB->get_in_or_equal($courseids, SQL_PARAMS_NAMED);
+
+        $params = array();
+        $params += $aparams;
+        $params += $cparams;
+
+        if($multipleassessment){
+            $sql = "SELECT DISTINCT i.itemname
+                    FROM `mdl_course_modules` cm 
+                    INNER JOIN `mdl_modules` m ON m.id = cm.module 
+                    INNER JOIN `mdl_grade_items` i on i.courseid = cm.course 
+                    INNER JOIN `mdl_grade_grades` g on g.itemid = i.id 
+                    WHERE m.name {$inactivities} 
+                    AND itemtype = 'mod' 
+                    AND g.finalgrade > 0  
+                    AND g.finalgrade IS NOT NULL 
+                    AND g.userid = $userid
+                    AND cm.course {$incourses}";
+        }
+        else {
+            $sql = "SELECT DISTINCT mcm.id, mcm.course, mcm.instance, mm.name
+                    FROM `mdl_course_modules` mcm
+                    JOIN `mdl_modules` mm ON mm.id = mcm.module
+                    WHERE mm.name {$inactivities}
+                    AND mcm.course {$incourses}";
+        }
+
+        if ($assessments = $DB->get_records_sql($sql, $params)){
+            return $assessments;
+        } else {
+            return $assessments;
+        }
+    }
+
+    public static function retrieve_submitted_assignment($userid){
+        global $DB;
+
+        $submittedassignment = array();
+
+        $sql = "SELECT * FROM `mdl_assign_submission` sub
+                INNER JOIN 	`mdl_assign` ass ON ass.id = sub.assignment 
+                AND userid = $userid AND status='submitted'";
+        
+        if ($submittedassignment = $DB->get_records_sql($sql)){
```

Comment:
Use Moodle's `$DB->count_records()` or `$DB->count_records_sql()`  instead of using `$DB->get_records_sql()`.